### PR TITLE
Some changes I needed to make to get this compiling in Xcode 6.3

### PR DESCRIPTION
**Pull request by nathankot**

### CODE REVIEW

**Comment by CodaFi** [comment]
Oh, thanks!  I've been waiting for Travis for so long I nearly abandoned this branch.  Anyhow, is there a chance you could also help fix some formatting stuff?  I see a few mistakes we've made that I can mark in line comments for you to fix.

File: Swiftx/Either.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ public enum Either<L, R> {
 
 	/// Case analysis for the Either type. If the value is Left(a), apply the first function to a;
 	/// if it is Right(b), apply the second function to b.
-	public func either<A>(#onLeft : L -> A, onRight : R -> A) -> A {
```
Comment (CodaFi) [inline Swiftx/Either.swift:54]: Why did you remove the labels here?

**Comment by nathankot** [comment]
@CodaFi sounds good :) It's late tonight so I will have a look at them tomorroW!

File: Swiftx/Result.swift [hunk 2]
```diff
@@ -36,7 +36,7 @@ public enum Result<V> {
 	/// Much like the ?? operator for Optional types, takes a value and a function,
 	/// and if the Result is Error, returns the error, otherwise maps the function over
 	/// the value in Value and returns that value.
-	public func fold<B>(value: B, f: V -> B) -> B {
+	public func fold<B>(value: B, _ f: V -> B) -> B {
```
Comment (CodaFi) [inline Swiftx/Result.swift]:
```
public func fold<B>(value : B, _ f : V -> B) -> B
```

File: Swiftx/Either.swift (hunk 1 above)
Comment (nathankot) [inline Swiftx/Either.swift:54]: I mistakenly remember seeing somewhere in the Swiftz docs using `either` with no labels (which was possible before 1.2) but I'll revert this :)

Comment (CodaFi) [inline Swiftx/Either.swift:54]: It's totally my fault if that's the case.  I'm trying to push for labels in the next release after this and accidentally included some in SwiftX.

**Comment by CodaFi** [comment]
Thanks again for this!

:boat:
